pymeasure/instruments/keithley/keithley2182.py

```diff
@@ -0,0 +1,399 @@
+        """Set active channel and configure channel for voltage measurement.
+
+        :param auto_range: Enables auto_range if True, else uses set voltage
+            range
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'VOLT';"
+                   f":SENS:VOLT:NPLC {nplc};")
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO 1")
+        self.check_errors()
+
+    def setup_temperature(self, nplc=5):
+        """Change active channel and configure channel for temperature measurement.
+
+        :param channel: Temperature measurement channel
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 50/60
+        """
+        self.write(":SENS:CHAN {ch};"
+                   ":SENS:FUNC 'TEMP';"
+                   f":SENS:TEMP:NPLC {nplc}")
+        self.check_errors()
+
+    def acquire_reference(self, func='VOLT'):
+        """Acquire a measurement and store it as the relative offset value.
+
+        Only acquires reference if offset state is ON.
+
+        :param func: 'VOLT' or 'TEMP'
+        """
+        self.write(f":SENS:{func}:""CHAN{ch}:REF:ACQ")
+
+
+class Keithley2182(KeithleyBuffer, Instrument):
+    """Represents the Keithley 2182 Nanovoltmeter and provides a
+    high-level interface for interacting with the instrument.
+
+    .. code-block:: python
+
+        keithley = Keithley2182("GPIB::1")
+
+        keithley.reset()                        # Return instrument settings to default values
+        keithley.thermocouple = 'S'             # Sets thermocouple type to S
+        keithley.active_channel = 1             # Sets channel 1 for active measurement
+        keithley.channel_function = 'voltage'   # Configures active channel for voltage measurement
+        print(keithley.voltage)                 # Prints the voltage in volts
+
+        keithley.ch_1.setup_voltage()           # Set channel 1 active and prepare voltage measurement
+        keithley.ch_2.setup_temperature()       # Set channel 2 active and prepare temperature measurement
+
+    """
+
+    def __init__(self, adapter, name="Keithley 2182 Nanovoltmeter",
+                 read_termination='\r', **kwargs):
+        super().__init__(adapter, name, read_termination=read_termination,
+                         includeSCPI=True, **kwargs)
+        self.auto_line_frequency()
+
+    ch_1 = Instrument.ChannelCreator(Keithley2182Channel, 1)
+    ch_2 = Instrument.ChannelCreator(Keithley2182Channel, 2)
+
+    #################
+    # Configuration #
+    #################
+
+    auto_zero = Instrument.control(
+        ":SYST:AZER:STAT?", ":SYST:AZER:STAT %d",
+        """Control the auto zero option.
+        Valid values are True (enabled) and False (disabled).""",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
```

Comment:
Here again, using `enabled` in the name, makes it self evident, without the need to write the values into the docstring:
```suggestion
    auto_zero_enabled = Instrument.control(
        ":SYST:AZER:STAT?", ":SYST:AZER:STAT %d",
        """Control the auto zero option (bool).""",
        validator=strict_discrete_set,
        values={True: 1, False: 0},
        map_values=True,
    )
```

Reply:
Thanks for the advice on boolean naming, I've updated the properties accordingly